java-client-core/src/main/java/io/sphere/sdk/client/SphereClientImpl.java

```diff
@@ -103,6 +104,15 @@ private SphereClientImpl(final SphereApiConfig config, final SphereAccessTokenSu
         exception.setSphereRequest(sphereRequest);
         exception.setUnderlyingHttpResponse(httpResponse);
         exception.setProjectKey(config.getProjectKey());
+
+        final List<String> errorMessagesDueToMappingError = asList("SearchPhaseExecutionException", "query_fetch", "RemoteTransportException", "SearchParseException", "search/phase/query+fetch");
+        httpResponse.getResponseBody()
+                .map(bytes -> bytesToString(bytes))
+                .map(body -> errorMessagesDueToMappingError.stream().anyMatch(errorWord -> body.contains(errorWord)) && body.toLowerCase().contains("product"))
+                .map(containsTerm -> {
+                    exception.addNote(String.format("Maybe it helps to reindex the products https://admin.sphere.io/%s/developers/danger but this may take a while.", config.getProjectKey()));
+                    return null;
```

Comment:
I just recovered this PR from my mailbox. What I wanted to say here is that I think you can use instead `ifPresent` to avoid returning something.